File: QueueHive_BE/src/main/java/queuehive/queuehive/service/CompanyService.java
package queuehive.queuehive.service;

import queuehive.queuehive.dto.CompanyDto;
import queuehive.queuehive.dto.CreateCompanyRequest;
import queuehive.queuehive.dto.UpdateCompanyRequest;

import java.util.List;

public interface CompanyService {
    CompanyDto registerCompany(CreateCompanyRequest request);
    CompanyDto approveCompany(Long companyId);
    CompanyDto rejectCompany(Long companyId); // New method
    void deleteCompany(Long companyId);
    List<CompanyDto> listApprovedCompanies();
    List<CompanyDto> listAllCompanies(); // Get all companies regardless of status
    List<CompanyDto> listPendingCompanies(); // New method
    CompanyDto getCompanyById(Long companyId); // New method
    CompanyDto getCompanyByOwnerId(Long ownerId); // Get company by owner ID
    CompanyDto updateCompany(Long companyId, UpdateCompanyRequest request);
    
    // New methods for analytics
    Integer getDailyVisitors(Long companyId);
    // This could return a more complex DTO for full stats
    List<String> getQueueStats(Long companyId); 
}

File: QueueHive_FE/queuehive/src/api/companyService.js
import http from './http';

const getApprovedCompanies = () => {
  return http.get('/companies');
};

const getServicesForCompany = (companyId) => {
  return http.get(`/services/company/${companyId}`);
};

const registerCompany = (companyData) => {
  return http.post('/companies', companyData);
};

const getCompanyByOwnerId = (ownerId) => {
  return http.get(`/companies/owner/${ownerId}`);
};

const getPendingCompanies = () => {
  return http.get('/companies/pending');
};

const approveCompany = (companyId) => {
  return http.put(`/companies/${companyId}/approve`);
};

const rejectCompany = (companyId) => {
  return http.put(`/companies/${companyId}/reject`);
};

const deleteCompany = (companyId) => {
  return http.delete(`/companies/${companyId}`);
};

const getCompanyById = (companyId) => {
  return http.get(`/companies/${companyId}`);
};

const getAllCompanies = () => {
  return http.get('/companies/all');
};

const companyService = {
  getApprovedCompanies,
  getAllCompanies,
  getServicesForCompany,
  registerCompany,
  getCompanyByOwnerId,
  getPendingCompanies,
  approveCompany,
  rejectCompany,
  deleteCompany,
  getCompanyById,
};

export default companyService;
